tests/cupy_tests/math_tests/test_rational.py

```diff
@@ -0,0 +1,38 @@
+import unittest
+
+import pytest
+
+import cupy
+from cupy import testing
+
+
+@testing.gpu
+class TestRational(unittest.TestCase):
+
+    @testing.for_int_dtypes(no_bool=True)
+    @testing.numpy_cupy_array_equal()
+    def test_gcd_basic(self, xp, dtype):
+        a = testing.shaped_random((6, 6), xp, dtype)
+        b = testing.shaped_random((6, 6), xp, dtype)
+        return xp.gcd(a, b)
+
+    @testing.for_float_dtypes()
```

Comment:
Add tests for Boolean and complex dtypes.
```suggestion
    @testing.for_dtypes(['?', 'e', 'f', 'd', 'F', 'D'])
```

Reply:
Done

Reply:
Add a test for bool dtype.

Reply:
The test for bool fails as mentioned below, even when I specifically add 'no' casting, this fails.
example
```
import numpy
import cupy

for xp in [cupy, numpy]:
    print(xp.__name__)
    a = xp.array([True, False, False])
    b = xp.array([True, False, True])
    print(xp.logaddexp(a, b, casting='no'))
```

> cupy
> [1.693  0.6934 1.313 ]
> numpy
> Traceback (most recent call last):
>   File "/D/work/Python/trial2.py", line 9, in <module>
>     print(xp.logaddexp(a, b, casting='no'))
> TypeError: ufunc 'logaddexp' not supported for the input types, and the inputs could not be safely coerced to any supported types according to the casting rule ''no''

Can you please tell me how else I should add this test because the proposed way typecasts it to int8 and the doesnt give the desired result.

Reply:
Please refer this line: https://github.com/cupy/cupy/blob/master/cupy/core/_routines_math.pyx#L744

Reply:
Thanks a lot, I have modified the ufunc and added the test case.